src/macaron/slsa_analyzer/git_url.py

```diff
@@ -268,20 +277,38 @@ def clone_remote_repo(clone_dir: str, url: str) -> Repo | None:
             )
             return None
 
+    # Ensure that the parent directory where the repo is cloned into exists.
+    parent_dir = Path(clone_dir).parent
+    parent_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
Why is creating the parent directories added here? Is it to be consistent with `Repo.clone_from`?

Reply:
Yes, it is.